src/pybind/mgr/cephadm/module.py

```diff
@@ -764,33 +751,33 @@ def serve(self):
             self._check_hosts()
 
             # refresh daemons
-            self.log.debug('refreshing daemons')
-            completion = self._get_daemons(maybe_refresh=True)
-            self._orchestrator_wait([completion])
-            # FIXME: this is a band-aid to avoid crashing the mgr, but what
-            # we really need to do here is raise health alerts for individual
-            # hosts that fail and continue with the ones that do not fail.
-            if completion.exception is not None:
-                self.log.error('failed to refresh daemons: %s' % completion.exception)
+            cutoff = datetime.datetime.utcnow() - datetime.timedelta(
+                seconds=self.daemon_cache_timeout)
+            cutoffs = cutoff.strftime(DATEFMT)
+            self.log.debug('refreshing daemons, cutoff %s' % cutoffs)
+            failures = []
+            for host, di in self.daemon_cache.items():
```

Comment:
wondering, if we should move the cutoff calculation into the loop, as querying a few hundred hosts might take some time